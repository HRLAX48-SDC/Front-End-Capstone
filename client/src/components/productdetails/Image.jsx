import React, {useState} from 'react';
import Carousel from 'react-bootstrap/Carousel';

function Image(props) {

  const [selectImg, setSelectImg] = useState(0);
  const [click, setClick] = useState(false);

  const handleClick = (index) => {
    setSelectImg(index);
    setClick(true);
  };

  return (
    <div className='row'>
      <div className='col-md-2'>
        {props.style.length ?
          props.style[props.styleNum].photos.map((img, index) => (
          <img
            className='p_s_img'
            key={index}
            src={img.thumbnail_url}
            onClick={() => handleClick(index)}
          ></img>)) : null}
      </div>
      <div className='col-md-10'>
        {props.style.length ?
        <Carousel variant='dark'>
          {click ?
          <img className='w-100 p_pic' src={props.style[props.styleNum].photos[selectImg].url}>
          </img>
          :
          props.style[props.styleNum].photos.map((img, index) => (
            <Carousel.Item key={index} interval={5000}>
              <img
                className='d-block w-100 p_pic'
                src={img.url}
              />
            </Carousel.Item>
          ))}
        </Carousel>
        :
        <div className='spinner-border' role='status'>
          <span className='visually-hidden'>Loading...</span>
        </div>
        }
      </div>
    </div>
  );
};

export default Image;