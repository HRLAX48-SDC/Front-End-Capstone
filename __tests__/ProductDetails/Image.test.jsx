import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { configure, mount, shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import Image from '../../client/src/components/productdetails/Image.jsx';

test('renders Image Gallery component correctly', () => {
  const style = [{
    style_id: 253620,
    name: 'Forest Green & Black',
    original_price: '140.00',
    sale_price: null,
    photos: [{
      thumbnail_url: "https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
      url: "https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80"
    }]
  }];
  const styleNum = 0;
  const expand = false;
  const imageGallery = renderer.create(<Image style={style} styleNum={styleNum} />).toJSON();
  expect(imageGallery).toMatchSnapshot();
});

test('Renders the status component', () => {
  const style = [{
    style_id: 253620,
    name: 'Forest Green & Black',
    original_price: '140.00',
    sale_price: null,
    photos: [{
      "thumbnail_url": "https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
      "url": "https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80"
    }, {
      "thumbnail_url": "https://images.unsplash.com/photo-1534011546717-407bced4d25c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
      "url": "https://images.unsplash.com/photo-1534011546717-407bced4d25c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2734&q=80"
    }, {
      "thumbnail_url": "https://images.unsplash.com/photo-1549831243-a69a0b3d39e0?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
      "url": "https://images.unsplash.com/photo-1549831243-a69a0b3d39e0?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2775&q=80"
    }, {
      "thumbnail_url": "https://images.unsplash.com/photo-1527522883525-97119bfce82d?ixlib=rb-1.2.1&auto=format&fit=crop&w=300&q=80",
      "url": "https://images.unsplash.com/photo-1527522883525-97119bfce82d?ixlib=rb-1.2.1&auto=format&fit=crop&w=668&q=80"
    }, {
      "thumbnail_url": "https://images.unsplash.com/photo-1556648202-80e751c133da?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
      "url": "https://images.unsplash.com/photo-1556648202-80e751c133da?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80"
    }, {
      "thumbnail_url": "https://images.unsplash.com/photo-1532543491484-63e29b3c1f5d?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
      "url": "https://images.unsplash.com/photo-1532543491484-63e29b3c1f5d?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1000&q=80"
    }, {
      "thumbnail_url": "https://images.unsplash.com/photo-1533779183510-8f55a55f15c1?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
      "url": "https://images.unsplash.com/photo-1533779183510-8f55a55f15c1?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80"
    }, {
      "thumbnail_url": "https://images.unsplash.com/photo-1560567546-4c6dbc16877b?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
      "url": "https://images.unsplash.com/photo-1560567546-4c6dbc16877b?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80"
    }, {
      "thumbnail_url": "https://images.unsplash.com/photo-1458253329476-1ebb8593a652?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
      "url": "https://images.unsplash.com/photo-1458253329476-1ebb8593a652?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80"
    }]
  }];
  const styleNum = 0;
  const expand = true;
  const wrapper = mount(<Image styleNum={styleNum} expand={expand} style={style} />);
  expect(wrapper).toContainMatchingElements(9,'img');

  const scrollUp = wrapper.find('.p_up');
  scrollUp.simulate('click');
  expect(wrapper).toContainMatchingElements(9,'img');

  const scrollDown = wrapper.find('.p_down');
  scrollDown.simulate('click');
  expect(wrapper).toContainMatchingElements(9,'img');

  const scrollRight = wrapper.find('.p_right').at(0);
  scrollRight.simulate('click');
  expect(wrapper).toContainMatchingElements(9,'img');

  const scrollLeft = wrapper.find('.p_left').at(0);
  scrollLeft.simulate('click');
  expect(wrapper).toContainMatchingElements(9,'img');
})
