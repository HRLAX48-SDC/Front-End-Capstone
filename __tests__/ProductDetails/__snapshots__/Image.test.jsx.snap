// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Image Gallery component correctly 1`] = `
<div
  className="row justify-content-around"
>
  <div
    className="col-md-2"
  >
    <button
      className="p_up"
      onClick={[Function]}
    >
      <i
        className="fas fa-chevron-up"
      />
    </button>
    <img
      className="p_s_img_highlight"
      onClick={[Function]}
      src="https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
    />
    <button
      className="p_down"
      onClick={[Function]}
    >
      <i
        className="fas fa-chevron-down"
      />
    </button>
  </div>
  <div
    className="col-md-9"
  >
    <div>
      <img
        alt="Forest Green & Black"
        className="p_pic"
        src="https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80"
      />
    </div>
  </div>
</div>
`;
